sources/packages/web/src/components/aest/StudentApplications.vue

```diff
@@ -0,0 +1,290 @@
+import CancelApplication from "@/components/students/modals/CancelApplicationModal.vue";
+import Tooltip from "primevue/tooltip";
+
+export default {
+  components: { Status, ConfirmEditApplication, CancelApplication },
+  directives: {
+    tooltip: Tooltip,
+  },
+  props: {
+    clientType: {
+      type: String,
+      required: true,
+    },
+    studentId: {
+      type: Number,
+      required: true,
+    },
+  },
+  setup(props: any) {
+    const studentDetail = ref({} as StudentDetail);
+    const loading = ref(false);
+    const applicationAndCount = ref({} as StudentApplicationAndCount);
+    const defaultSortOrder = -1;
+    const currentPage = ref();
+    const currentPageLimit = ref();
+    const { dateString } = useFormatters();
+    const router = useRouter();
+    const hasRestriction = ref(false);
+    const editApplicationModal = ref({} as ModalDialog<boolean>);
+    const programYear = ref({} as ProgramYearOfApplicationDto);
+    const showModal = ref(false);
+    const selectedApplicationId = ref(0);
+
+    /**
+     * function to load applicationListAndCount respective to the client type
+     * @param page page number, if nothing passed then DEFAULT_PAGE_NUMBER
+     * @param pageCount page limit, if nothing passed then DEFAULT_PAGE_LIMIT
+     * @param sortField sort field, if nothing passed then StudentApplicationFields.status
+     * @param sortOrder sort oder, if nothing passed then DataTableSortOrder.ASC
+     */
+    const getStudentApplications = async (
+      page = DEFAULT_PAGE_NUMBER,
+      pageCount = DEFAULT_PAGE_LIMIT,
+      sortField = StudentApplicationFields.Status,
+      sortOrder = DataTableSortOrder.ASC,
+    ) => {
+      switch (props.clientType) {
+        case ClientIdType.Student:
+          applicationAndCount.value = await StudentService.shared.getAllStudentApplications(
+            page,
+            pageCount,
+            sortField,
+            sortOrder,
+          );
+          break;
+        case ClientIdType.AEST:
+          applicationAndCount.value = await ApplicationService.shared.getAllApplicationsForStudent(
+            props.studentId,
+            page,
+            pageCount,
+            sortField,
+            sortOrder,
+          );
+          break;
+      }
+    };
+
+    const reloadApplication = async () => {
+      await getStudentApplications();
+    };
+
+    onMounted(async () => {
+      if (props.clientType === ClientIdType.Student) {
+        const restrictions = await StudentService.shared.getStudentRestriction();
+        hasRestriction.value = restrictions.hasRestriction;
+      }
+      reloadApplication();
+    });
+
+    // pagination sort event callback
+    const paginationAndSortEvent = async (event: any) => {
+      loading.value = true;
+      currentPage.value = event?.page;
+      currentPageLimit.value = event?.rows;
+      await getStudentApplications(
+        event?.page,
```

Comment:
Eventually I feel page and rows will be always present in the call, Can we please remove the ?.

Reply:
mm, actually event will not be undefined for sort and pagination datatable callback, you are correct